http/client/client.go

```diff
@@ -29,41 +24,14 @@ type (
 	}
 
 	tokenSupplier interface {
-		GetToken() (string, error)
+		GetToken(ctx context.Context) (string, error)
 	}
 )
 
-// NewRoundTripper creates an http.RoundTripper that propagates OpenTelemetry trace headers.
```

Comment:
I had to move this stuff out because there was a circular dependency (authenticated HTTP client requires OIDC client which required the base unauthenticated HTTP client).